battleship-client/src/main/java/com/academy/solid/nie/client/communication/SocketClient.java

```diff
@@ -27,11 +27,12 @@
 
     @Override
     public void run() throws IOException {
-        String fromServer;
-        while (!("Provide ships").equals(fromServer = in.readLine())) {
+        String fromServer = in.readLine();
```

Comment:
Just to make sure: which logic you are after here - the one where `readLine` is called once or where each iteration calls it?

Reply:
We avoid inline assignment in line 31. So we moved iteration calls to end of the while loop.

Reply:
OK thanks.